google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -121,6 +122,15 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket pb) {
     public String getValue() {
       return value;
     }
+
+    public static PublicAccessPrevention parse(String value) {
+      String upper = value.toUpperCase();
+      try {
+        return valueOf(upper);
+      } catch (IllegalArgumentException ignore) {
```

Comment:
Do we still need to catch this exception? Where is it being thrown from now?

Reply:
If the string passed to valueOf doesn't match one of the defined enum values, this exception is thrown. We catch it and return unknown so that we don't have to re-implement the traversal and testing logic already implemented in valueOf.